lib/services/annil/annil.dart

```diff
@@ -16,15 +17,27 @@ import 'package:collection/collection.dart';
 
 class AnnilService extends ChangeNotifier {
   final Ref ref;
-  final Dio _client = Dio();
+  final Dio client = Dio();
 
   List<LocalAnnilServer> servers = [];
   Map<int, String?> etags = {};
   List<String> albums = [];
 
   AnnilService(this.ref) {
-    _client.interceptors.add(RetryInterceptor(
-      dio: _client,
+    final config = ref.read(settingsProvider);
+    if (config.skipCertificateVerification.value) {
```

Comment:
`config.skipCertificateVerification.value` may change.

Reply:
I'm not familiar with flutter. Would `if (ref.read(settingsProvider).skipCertificateVerification.value) {}` work?

Reply:
Or pass `Ref` to `DownloadTask`?

Reply:
I think you should not use `config.skipCertificateVerification.value` with `if` statement here, but return an expression with `config.skipCertificateVerification.value` instead of `true`/`false`.

Reply:
updated